### PR TITLE
Add new mvi core implementation 

### PR DESCRIPTION
**Pull request by Hukumister**

### CODE REVIEW

File: mvi-core/src/main/java/ru/coderedwolf/mvi/core/AbstractStore.kt [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    private val actionProcessor = PublishProcessor.create<Action>()
+    private val effectProcessor = PublishProcessor.create<Effect>()
+    private val eventProcessor = BehaviorProcessor.create<Event>()
+
+    init {
+        Flowables.zip(
+            stateProcessor,
+            effectProcessor
+        )
+            .scan(initialState) { _, (state, effect) -> reducer.invoke(state, effect) }
```
Comment (kdk96) [inline mvi-core/src/main/java/ru/coderedwolf/mvi/core/AbstractStore.kt]: ignore argument in scan ? whaat

File: mvi-core/src/main/java/ru/coderedwolf/mvi/core/AbstractStore.kt [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    private val reducer: Reducer<State, Effect>,
+    private val middleware: Middleware<Action, State, Effect>,
+    bootstrapper: Bootstrapper<Action>? = null,
+    private val eventProducer: EventProducer<State, Effect, Event>? = null
+) : Store<Action, State, Event> {
+
+    private val compositeDisposable = CompositeDisposable()
+
+    private val stateProcessor = BehaviorProcessor.createDefault(initialState)
+    private val viewStateProcessor = BehaviorProcessor.create<State>()
```
Comment (kdk96) [inline mvi-core/src/main/java/ru/coderedwolf/mvi/core/AbstractStore.kt]: so, this processor should be deleted

File: mvi-core/src/main/java/ru/coderedwolf/mvi/core/AbstractStore.kt (hunk 1 above)
Comment (kdk96) [reply mvi-core/src/main/java/ru/coderedwolf/mvi/core/AbstractStore.kt]: I have idea. Maybe we should to create wrappers for reducer, and middleware, where we will catch errors and transfer them to onError method? in reducer wrapper we will return previous state, in middleware wrapper - empty flowable

File: mvi-core/src/main/java/ru/coderedwolf/mvi/core/elements/Middleware.kt [hunk 3]
```diff
@@ -3,6 +3,6 @@ package ru.coderedwolf.mvi.core.elements
 import io.reactivex.Flowable
 
 /**
- * @author CodeRedWolf. Date 13.10.2019.
+ * @author HaronCode.
  */
 typealias Middleware<Action, State, Effect> = (action: Action, state: State) -> Flowable<Effect>
```
Comment (kdk96) [inline mvi-core/src/main/java/ru/coderedwolf/mvi/core/elements/Middleware.kt:8]: I think we can replace Flowable with Publisher

**Comment by Hukumister** [comment]
Close request because move this module to external library

**Pull request closed without merging**